app/src/main/java/com/celzero/bravedns/ui/PauseActivity.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2021 RethinkDNS and its authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.celzero.bravedns.ui
+
+import android.content.Context
+import android.content.Intent
+import android.content.res.Configuration
+import android.os.Bundle
+import android.os.SystemClock
+import android.view.MotionEvent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import by.kirich1409.viewbindingdelegate.viewBinding
+import com.celzero.bravedns.R
+import com.celzero.bravedns.automaton.FirewallManager
+import com.celzero.bravedns.databinding.PauseActivityBinding
+import com.celzero.bravedns.service.BraveVPNService
+import com.celzero.bravedns.service.PersistentState
+import com.celzero.bravedns.service.VpnController
+import com.celzero.bravedns.util.Constants.Companion.PAUSE_VPN_EXTRA_MILLIS
+import com.celzero.bravedns.util.Constants.Companion.INIT_TIME_MS
+import com.celzero.bravedns.util.Themes
+import com.celzero.bravedns.util.Utilities.Companion.humanReadableTime
+import kotlinx.coroutines.*
+import org.koin.android.ext.android.inject
+import java.util.concurrent.TimeUnit
+
+class PauseActivity : AppCompatActivity(R.layout.pause_activity) {
+    private val b by viewBinding(PauseActivityBinding::bind)
+    private val persistentState by inject<PersistentState>()
+    @Volatile var j: CompletableJob? = null
+
+    enum class AutoOp {
+        INCREMENT, DECREMENT, NONE
+    }
+
+    @Volatile var autoOp = AutoOp.NONE
+    var lastStopActivityInvokeTime: Long = INIT_TIME_MS
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        setTheme(Themes.getCurrentTheme(isDarkThemeOn(), persistentState.theme))
+        super.onCreate(savedInstanceState)
+    }
+
+    private fun initView() {
+        FirewallManager.getApplistObserver().observe(this, {
+            val blockedList = it.filter { a -> !a.isInternetAllowed }
+            b.pacTimerDesc.text = getString(R.string.pause_desc, blockedList.size.toString())
+        })
+    }
```

Comment:
see: https://developer.android.com/topic/libraries/architecture/coroutines#restart

Reply:
`observeTimer` is called from onCreate